### ISSUE TITLE
Adding a few non-cmor variables for the RESCUE project

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#835]
For the RESCUE project a few new non-cmor variables need to be added: three variables produced by LPJ-GUESS and two by the co2box model.

The `EC-Earth3-ESM-1` configuration will be used and for a (first) start the optimesm request will be used but with the extension of this five variables (names have still to be determined). All five variables will be yearly output.

For RESCUE a new option to LPJ-GUESS and the co2box component are added in the exising `EC-Earth3-ESM-1` configuration in order to enable CO2 capture techniques modelling.

For LPJG the correct .ins files should be generated bu `genecec` as well.

**Comment by nierad** [comment]
Hi @treerink : 
Thanks a lot!
This is the branch: https://github.com/login?return_to=https%3A%2F%2Fgithub.com%2FEC-Earth%2Fece2cmor3%2Fissues%2F835

**Comment by treerink** [comment]
The `EC-Earth3-ESM-1` RESCUE branch:
 https://svn.ec-earth.org/ecearth3/branches/projects/rescue
and corresponding [ECE portal issue 1377](https://dev.ec-earth.org/issues/1377).

**Comment by treerink** [comment]
Ready for testing. See also the updated rescue branch in the ECE repository for the new rescue `output-control-files`.

**Comment by treerink** [comment]
@nierad I think all is in place. The three new LPJG variables are added to the already in optimesm created LPJGyr table and the two new co2box variables are added to the Eyr table. Please check the variable metadata which is created in the `add-rescue-variables.sh` script, that is the place to adjust. But it can be viewed also in the cmor tables once you just run the `add-rescue-variables.sh` script.

Before cmorising rescue data one also need to run the `add-rescue-variables.sh` script.

**Comment by nierad** [comment]
Hi @treerink : 
Thanks a lot again! I will test asap.

**Comment by etiennesky** [comment]
hi @treerink @nierad I don't think that these variables should go to the LPJGyr table, because other modelling centers will be producing these variables as well. We don't have consensus yet on which table to use though.

Also, I think a few variables (fBECCS, fBECCSatm) are with monthly frequency but this should be yearly instead?

**Comment by treerink** [comment]
@nierad @etiennesky Ok I can adjust the table name and also the frequency as soon you have consensus. I had the impression that at least something working is appreciated on very short notice, such that experiments can start and we can adjust it later if necessary? Anyway let me know and I will adjust it.

**Comment by etiennesky** [comment]
> hi @treerink @nierad I don't think that these variables should go to the LPJGyr table, because other modelling centers will be producing these variables as well. We don't have consensus yet on which table to use though.
> 
> Also, I think a few variables (fBECCS, fBECCSatm) are with monthly frequency but this should be yearly instead?

Hi @treerink 

1) the problem is : "fBECCS" and "fBECCSatm" have   "frequency": "mon" / 

2)
we need to have the experiment as esm-hist (not historical) with parent esm-piControl (not piControl), since these are emission-driven experiments - same as in OptimESM which is described here  https://dev.ec-earth.org/issues/1359

3) we have not finalized the variables yet with the other RESCUE partners, will update as soon as possible

**Comment by treerink** [comment]
Concerning point 2: I have adjusted the `experiment_id` and `parent_experiment_id` for both the OprimESM and RESCUE case (This can be also fixed in older data with either the `cmor-fixer` of the `cmorMDfixer`). Unfortunately I was not aware of issue 1359, I did not receive any email alert via the portal, just saw it here from your link.

**Comment by etiennesky** [comment]
> Concerning point 2: I have adjusted the `experiment_id` and `parent_experiment_id` for both the OprimESM and RESCUE case (This can be also fixed in older data with either the `cmor-fixer` of the `cmorMDfixer`). Unfortunately I was not aware of issue 1359, I did not receive any email alert via the portal, just saw it here from your link.

thanks! I think you are a watcher of issue 1359. Please can you fix the filenames as well ?

**Comment by klauswyser** [comment]
Aren't filenames automatically created from the ``experiment_id``?

If you have already processed data with a wrong ``experiment_id`` you probably have to run a bash script to rename the files (in addition to cmor[MD]fixer). It would be nice if this could also be included in the cmor- or cmorMDfixer but not sure if this is feasible. How efficient is python when it comes to filesystem operations such as renaming files and directories?

**Comment by etiennesky** [comment]
> Aren't filenames automatically created from the `experiment_id`?
> 
> If you have already processed data with a wrong `experiment_id` you probably have to run a bash script to rename the files (in addition to cmor[MD]fixer). It would be nice if this could also be included in the cmor- or cmorMDfixer but not sure if this is feasible. How efficient is python when it comes to filesystem operations such as renaming files and directories?

I am more concerned about what is in the output control files than the cmor fixer. The filenames should be consistent with the actual `experiment_id`, and this will be needed for future runs.

**Comment by treerink** [comment]
> > Aren't filenames automatically created from the `experiment_id`?
> > If you have already processed data with a wrong `experiment_id` you probably have to run a bash script to rename the files (in addition to cmor[MD]fixer). It would be nice if this could also be included in the cmor- or cmorMDfixer but not sure if this is feasible. How efficient is python when it comes to filesystem operations such as renaming files and directories?
> 
> I am more concerned about what is in the output control files than the cmor fixer. The filenames should be consistent with the actual `experiment_id`, and this will be needed for future runs.

But that part should be fine now (after updating the rescue branch), have you experienced a problem?

**Comment by etiennesky** [comment]
> But that part should be fine now (after updating the rescue branch), have you experienced a problem?

what I wrote in the issue, is that the filename should follow the experiment_id, so they should be calld esm-hist not historical. This gives us problems when we are looking for the files for the esm-hist experiment, they are not found

**Comment by etiennesky** [comment]
> @etiennesky : have you cmorised any results with the updated metadata template? In that case the filename should include "esm-hist" and not "historical" because the DRS (incl. filename) is created automatically from the metadata.

yes, @treerink cmorization is fine and the netcdf files have esm-hist in their names. I think I haven't explained the problem well. 

The metadata files in the output control files  for optimesm should have esm-hist not historical in the names e.g.  https://dev.ec-earth.org/projects/ecearth3/repository/entry/ecearth3/branches/projects/optimesm/runtime/classic/ctrl/output-control-files/optimesm/metadata-cmip6-CMIP-historical-EC-EARTH-ESM-1-ifs-template.json should be named metadata-cmip6-CMIP-esm-hist-EC-EARTH-ESM-1-ifs-template.json  not metadata-cmip6-CMIP-historical-EC-EARTH-ESM-1-ifs-template.json 

In order to make things work correctlly we had to rename the json files. Thanks!

**Comment by treerink** [comment]
Ok, clear now, indeed that I hadn't noticed. Will fix that.

**Comment by treerink** [comment]
The metadata files with the `esm-hist` label have been commited to the `output-control-files` in thei optimesm and rescue ECE branches.

**Comment by treerink** [comment]
@etiennesky now to point 1:
`the problem is : "fBECCS" and "fBECCSatm" have "frequency": "mon" /`
Should I change them to yearly variables? I see still a questionmark, but if this is sorted out, then I can adjust it. Can you confirm?

**Comment by etiennesky** [comment]
> @etiennesky now to point 1: `the problem is : "fBECCS" and "fBECCSatm" have "frequency": "mon" /` Should I change them to yearly variables? I see still a questionmark, but if this is sorted out, then I can adjust it. Can you confirm?
there is no need to rush this, when I know the final variables I will communicate. but yes, they are all yearly variables.

My main point is you already put them in the yearly table, but the frequency is monthly, so that's a bug.

**Comment by treerink** [comment]
In c3d7ffc the ISM component have been removed from the `source_type` for the OptimESM & RESCUE cases.

The corrected output-control files have been commited to the ECE repository branches.

See also [#839](https://github.com/EC-Earth/ece2cmor3/issues/839#issuecomment-2595630598).
...

**Comment by etiennesky** [comment]
Hi @nierad when looking at the cDACCS and fDACCS variables they are inconsistent, cDACCS has units "kg", dimension "time" whereas fDACCS has  "units": "1e-06", and "dimensions": "longitude latitude time",

rather I think this should be 

cDACCS units kg/m2 dimensions lon lat time

fDACCS units kg/m2/s dimensions lon lat time

what do you think?

**Comment by nierad** [comment]
These variables should be stored by the boxmodel and, therefore, the unit "kg" for cDACCS seems about right. fDACCS should, thus, be kg/s.

Re dimensions: I don't know which one is officially correct (lat or latitude). Both works with cdo at least. The time-dimension, imo,  should be "time" for fDACCS and "time1" (yrPt) for cDACCS.

I think the settings in  
./ece2cmor3/scripts/add-rescue-variables.sh
are from when we thought we compute the xDACCS varoables in LPJG and onlylater decided to let the co2box-model deal with it, no?

**Comment by treerink** [comment]
I will have a look to the  `add-rescue-variables.sh` script. I think we shaped it at that moment to our best knowledge, no problem to correct or to adjust according to the latest preferences. Let's see whether we can get this into a good shape. 

For RESCUE also CMIP6Plus  is used for the most like for OptimESM I guess? Or do I miss here something, and is this different?

**Comment by nierad** [comment]
Hi @treerink, 
Yes, RESCUE is built on OptimESM.
